tools/sst_dump_tool.cc

```diff
@@ -217,9 +222,35 @@ int SstFileDumper::ShowAllCompressionSizes(
           imoptions, moptions, ikc, &block_based_table_factories, i.first,
           0 /* sample_for_compression */, compress_opt,
           false /* skip_filters */, column_family_name, unknown_level);
-      uint64_t file_size = CalculateCompressedTableSize(tb_opts, block_size);
-      fprintf(stdout, "Compression: %s", i.second);
-      fprintf(stdout, " Size: %" PRIu64 "\n", file_size);
+      uint64_t num_data_blocks = 0;
+      uint64_t file_size = CalculateCompressedTableSize(tb_opts, block_size,
+                                                        &num_data_blocks);
+      fprintf(stdout, "Compression: %-24s", i.second);
+      fprintf(stdout, " Size: %10" PRIu64, file_size);
+      fprintf(stdout, " Blocks: %6" PRIu64, num_data_blocks);
+      const uint64_t compressed_blocks =
+        opts.statistics->getAndResetTickerCount(NUMBER_BLOCK_COMPRESSED);
+      const uint64_t not_compressed_blocks =
+        opts.statistics->getAndResetTickerCount(NUMBER_BLOCK_NOT_COMPRESSED);
+      // When the option enable_index_compression is true,
+      // NUMBER_BLOCK_COMPRESSED is incremented for index block(s).
+      if( (compressed_blocks + not_compressed_blocks) > num_data_blocks ){
+        num_data_blocks = compressed_blocks + not_compressed_blocks;
+      }
+      const uint64_t ratio_not_compressed_blocks =
+        (num_data_blocks - compressed_blocks) - not_compressed_blocks;
+      const double compressed_pcnt = ( 0 == num_data_blocks ) ? 0.0 :
+        ( ( (double)compressed_blocks / (double)num_data_blocks ) * 100.0 ) ;
+      const double ratio_not_compressed_pcnt = ( 0 == num_data_blocks ) ? 0.0 :
+        ( ( (double)ratio_not_compressed_blocks / (double)num_data_blocks ) * 100.0 ) ;
+      const double not_compressed_pcnt = ( 0 == num_data_blocks ) ? 0.0 :
+        ( ( (double)not_compressed_blocks / (double)num_data_blocks ) * 100.0 ) ;
```

Comment:
We follow Google C++ style, which requires us to use C++ style casting instead: https://google.github.io/styleguide/cppguide.html#Casting